packages/core/contracts/Escrow.sol

```diff
@@ -229,26 +225,48 @@ contract Escrow is IEscrow, ReentrancyGuard {
         if (remainingFunds > 0) {
             _safeTransfer(token, launcher, remainingFunds);
             remainingFunds = 0;
+            reservedFunds = 0;
+        }
+        if (status == EscrowStatuses.ToCancel) {
+            status = EscrowStatuses.Cancelled;
+            emit Cancelled();
+        } else {
+            status = EscrowStatuses.Complete;
+            emit Completed();
         }
-        status = EscrowStatuses.Complete;
-        emit Completed();
     }
 
     function storeResults(
```

Comment:
For backwards compatibility keep a storeResults without funds reserve

Reply:
If no amount is passed the status has to be Pending or Partial.
If the ammount is passed the status has to be to cancel